src/main/java/address/model/datatypes/ReadableViewablePerson.java

```diff
@@ -0,0 +1,18 @@
+package address.model.datatypes;
+
+import address.model.ModelManager;
+import javafx.collections.ObservableList;
+
+import java.util.Collection;
+import java.util.function.Supplier;
+
+/**
+ * Same purpose as {@link ReadablePerson}, extended with additional status data viewable by user.
+ */
+public interface ReadableViewablePerson extends ReadablePerson {
```

Comment:
I think `ViewableReadablePerson` is a better name. `ReadableViewablePerson` and `ReadablePerson` suggest that we have both `ViewablePerson` and `Person` that will be processed separately. While in this case, we have `ReadablePerson` with `Viewable` property, so `ViewableReadablePerson` suggests that the base object is `ReadablePerson`.